WordPress/gutenberg
# Migrating `BlockPatternSetup` to use updated `Composite` implementation

## What?

This PR updates [`BlockPatternSetup` in `@wordpress/block-editor`](https://github.com/WordPress/gutenberg/blob/72270c155e0518e89828e76145625fafdcd3ace1/packages/block-editor/src/components/block-pattern-setup/index.js) (and related sub-components) to use the updated `Composite` implementation from #54225.

Additionally, it reworks some of the UX to improve the keyboard/AT experience.


## Why?

In #54225, an updated implementation of `Composite` was added to `@wordpress/components`. As per #55224, all consumers of `Composite` need to migrate from the old version to the new version.


## How?

 - Removes `__unstableComposite` imports from `@wordpress/components`
 - Adds private `Composite*` exports from `@wordpress/components`
 - Refactors `BlockPatternSetup` and subcomponents to use updated `Composite` components


## Testing Instructions

Unfortunately `BlockPatternSetup` is no longer used in Gutenberg (see #52374), but because it is [used by various plugins](https://wpdirectory.net/search/01H4NNPAC2N1MBYKW7XZEA5SW2) ([despite being marked as 'experimental'](https://github.com/WordPress/gutenberg/blob/72270c155e0518e89828e76145625fafdcd3ace1/packages/block-editor/src/components/index.js#L158)!), we need to keep it around.

With the JetPack plugin installed, add a JetPack forms block to a page/post, and open the "Form Patterns" dialog.

<p align="center">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/f751a0c2-96bd-4a75-8f74-699b2aca39d6" alt="An empty JetPack forms block, showing various template options, and a 'Explore Form Patterns' button, circled in red." width="80%">
</p>

The (default) grid view should present as a single tab stop, with arrow keys moving up and down through the various options, as before. (This somewhat confusingly doesn't behave as a grid, but that's a problem outside of the scope of this PR!)

<p align="center">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/8b882ca4-b1f3-409c-9acd-39d30fecc0d8" alt="A 'Choose a pattern' dialog, showing previews of several form patterns." width="80%">
</p>

The carousel view should also behave as before, with the arrow buttons at the bottom navigating through the different patterns.

<p align="center">
  <img src="https://github.com/WordPress/gutenberg/assets/159848/f8dce243-cdd4-4af7-985c-163878596a0d" alt="A 'Choose a pattern' dialog, showing a large preview of a single form pattern." width="80%">
</p>
